.github/workflows/build-test-distribute.yaml

```diff
@@ -0,0 +1,161 @@
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: go.mod
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.kuma-dev
+          key: ${{ runner.os }}-devtools-${{ hashFiles('mk/dependencies/deps.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-devtools
+      - name: Free up disk space for the Runner
+        run: |
+          # source: https://github.com/apache/flink/blob/master/tools/azure-pipelines/free_disk_space.sh
+          echo "=============================================================================="
+          echo "Freeing up disk space on CI system"
+          echo "=============================================================================="
+          echo "Listing 100 largest packages"
+          dpkg-query -Wf '${Installed-Size}\t${Package}\n' | sort -n | tail -n 100
+          df -h
+          echo "Removing large packages"
+          sudo apt-get remove -y '^ghc-8.*' || true
+          sudo apt-get remove -y '^dotnet-.*' || true
+          sudo apt-get remove -y '^llvm-.*' || true
+          sudo apt-get remove -y 'php.*' || true
+          sudo apt-get remove -y azure-cli google-cloud-sdk hhvm google-chrome-stable firefox powershell mono-devel  || true
+          sudo apt-get autoremove -y
+          sudo apt-get clean
+          echo "Removing large directories"
+          sudo rm -rf /usr/share/dotnet /usr/local/lib/android /opt/ghc
+          docker system prune --all -f
+          sudo df -h
+      - name: Run tests
+        run: |
+          make test TEST_REPORTS=1
+      - name: Save test reports
+        uses: actions/upload-artifact@v2
+        with:
+          name: test-reports
+          path: build/reports
+          retention-days: 30
+      - name: Cleanup build folder
+        run: |
+          rm -rf ./build
+      - uses: actions/download-artifact@v2
+        if: ${{ ! contains(github.event.pull_request.labels.*.name, 'ci/skip-container-structure-test') }}
+        with:
+          name: build-output
+      - name: Unpack artifact
+        run: |
+          tar -xzf build.tar.gz
+      - name: Prepare for container structure test
+        if: ${{ ! contains(github.event.pull_request.labels.*.name, 'ci/skip-container-structure-test') }}
+        run: |
+          sudo apt-get update; sudo apt-get install -y qemu-user-static binfmt-support
+      - name: Run container structure test
+        if: ${{ ! contains(github.event.pull_request.labels.*.name, 'ci/skip-container-structure-test') }}
+        run: |
+          make test/container-structure
+  distributions:
+    needs: test
+    # this jobs will still run even if "test" is skipped
+    if: ${{ always() }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Check if force push"
+        if: ${{ contains(github.event.pull_request.labels.*.name, 'ci/force-publish') }}
```

Comment:
```suggestion
        # TODO expand to also do this when job runs on protected branch or a tag
        if: ${{ contains(github.event.pull_request.labels.*.name, 'ci/force-publish') }}
```

Reply:
You mean, we don't ask for the label when running on a protected branch or a tag?

Reply:
Yeah on a protected branch or a tag we systematically publish (though right now you shouldn't do this because there are no credentials to publish in the github repo)